eos-updater/fetch.c

```diff
@@ -468,7 +468,7 @@ static FlatpakUpdateFlags
 update_flags_for_action_flags (EuuFlatpakRemoteRefActionFlags action_flags)
 {
   return !(action_flags & EUU_FLATPAK_REMOTE_REF_ACTION_FLAG_IS_DEPENDENCY) ?
-      FLATPAK_UPDATE_FLAGS_NO_DEPLOY :
```

Comment:
IIRC the logic here is that before reboot, we fully install/update all dependencies and do a NO_DEPLOY update for the actual ref we need to install during the next boot, where we will do a subsequent NO_PULL update to complete it. (https://github.com/endlessm/eos-updater/blob/master/libeos-updater-flatpak-installer/perform-flatpak-actions.c#L166 and https://github.com/endlessm/eos-updater/blob/master/libeos-updater-flatpak-installer/perform-flatpak-actions.c#L52)

In other words, I am pretty sure that this code path is only hit during the pre-reboot preparation, and you should instead add the NO_PRUNE options to other call sites mentioned above, since that's the code that runs at boot.
(That's also what makes me think that the upstream logic is buggy in the other PR.)

Reply:
We talked about this in London today, and we concluded that the prune operations which flatpak does should be conditional on `!NO_PULL`, rather than anything to do with `NO_DEPLOY`. flatpak’s pruning is based on reachability from refs, and deploying/not deploying an app doesn’t affect its refs. Pulling/Not pulling an app does.

Reply:
This likely needs fixing in other places in flatpak too.

Reply:
Yes, I think you're right. But also, talking to @pwithnall he seems to be happy on not pruning on here either, so I think I'll keep setting a `NO_PRUNE` flag here anyway, regardless of whether it's not really run at boot time.